go-controller/pkg/ovn/egressip.go

```diff
@@ -416,47 +537,27 @@ func (e *egressIPMode) createEgressPolicy(podIps []net.IP, status egressipv1.Egr
 		} else if !isEgressIPv6 && !utilnet.IsIPv6(podIP) {
 			filterOption = fmt.Sprintf("ip4.src == %s", podIP.String())
 		}
-		policyIDs, err := findLRPolicyIDs(filterOption, egressIPName)
+		policyIDs, err := findReroutePolicyIDs(filterOption, egressIPName, gatewayRouterIP)
 		if err != nil {
 			return err
 		}
 		if policyIDs == nil {
-			if packetMark != 0 {
-				_, stderr, err = util.RunOVNNbctl(
-					"--id=@lr-policy",
-					"create",
-					"logical_router_policy",
-					"action=reroute",
-					fmt.Sprintf("match=\"%s\"", filterOption),
-					fmt.Sprintf("priority=%v", egressIPReroutePriority),
-					fmt.Sprintf("nexthop=%s", gatewayRouterIP),
-					fmt.Sprintf("external_ids:name=%s", egressIPName),
-					fmt.Sprintf("options:pkt_mark=%v", packetMark),
-					"--",
-					"add",
-					"logical_router",
-					ovnClusterRouter,
-					"policies",
-					"@lr-policy",
-				)
-			} else {
-				_, stderr, err = util.RunOVNNbctl(
-					"--id=@lr-policy",
-					"create",
-					"logical_router_policy",
-					"action=reroute",
-					fmt.Sprintf("match=\"%s\"", filterOption),
-					fmt.Sprintf("priority=%v", egressIPReroutePriority),
-					fmt.Sprintf("nexthop=%s", gatewayRouterIP),
-					fmt.Sprintf("external_ids:name=%s", egressIPName),
-					"--",
-					"add",
-					"logical_router",
-					ovnClusterRouter,
-					"policies",
-					"@lr-policy",
-				)
-			}
+			_, stderr, err = util.RunOVNNbctl(
+				"--id=@lr-policy",
+				"create",
+				"logical_router_policy",
+				"action=reroute",
+				fmt.Sprintf("match=\"%s\"", filterOption),
+				fmt.Sprintf("priority=%v", egressIPReroutePriority),
+				fmt.Sprintf("nexthop=%s", gatewayRouterIP),
```

Comment:
@alexanderConstantinescu  just trying to understand the code, does this still route the egress to GR, not mp0? or is it to force egress traffic to the egress ip node via the GR?

Reply:
>  does this still route the egress to GR, not mp0?

Correct

> or is it to force egress traffic to the egress ip node via the GR?

Correct. So, we need to send it to GR so that the SNAT rules can be applied before the packet exists the egress node

Reply:
I was mislead by the comment where I thought the `Traffic now` was referring to the current egressip traffic. all good now.
```
Traffic now exists the cluster node though ovn-k8s-mp0 which means that for new local gateway mode all we need is to create a logical router policy forcing south -> north traffic from a pod to the assigned egress node, and then apply a NAT rule on that node as to snat to the egress IP.
```